web/src/project/ProjectSummary.tsx

```diff
@@ -1,183 +1,364 @@
+                replace: true,
+            });
+            _setPageLimit(parseInt(value));
+            setPageNumber(1);
+        },
+        [projectName, pageLimit, page]
+    );
 
     const _updateProjectSummary = React.useCallback(() => {
-        if (_.isUndefined(suppliedDirectToken)) {
-            setSuppliedDirectToken(!!directToken)
-        }
-        getProjectSummary(directToken, false)
-    }, [getProjectSummary, suppliedDirectToken, directToken])
+        getProjectSummary(directToken);
+    }, [getProjectSummary, directToken]);
 
     // retrigger if project changes, or pageLimit changes
-    React.useEffect(
-        _updateProjectSummary,
-        [globalContext.project, pageLimit]
-    )
-
-    let table: React.ReactElement = <></>
-
-
-    const pageNumber = pageTokens.length + 1
-    const totalPageNumbers = Math.ceil((summary?.total_samples || 0) / pageLimit)
-
-    let pageOptions: React.ReactElement = <></>
-    if (suppliedDirectToken) {
-        let supplied_samples = summary?.participants.reduce((p, s) => s.samples.length + p, 0)
-        pageOptions = <>
-            {<Button disabled={isLoading} onClick={(async () => {
-                await getProjectSummary(undefined, false)
-                setSuppliedDirectToken(false)
-                searchParams.delete('token')
-                setSearchParams(searchParams)
-            })}>Go to start</Button>}
-            <span style={{ padding: "0 10px" }}>({supplied_samples} / {summary?.total_samples} samples)</span>
-        </>
-    } else {
+    React.useEffect(_updateProjectSummary, [projectName, pageLimit, page]);
 
-        pageOptions = <>
-            {pageTokens.length >= 1 && <Button disabled={isLoading} onClick={(async () => {
-                setPageTokens(pageTokens.slice(undefined, pageTokens.length - 1))
-                await getProjectSummary(pageTokens[pageTokens.length - 2], false)
-            })}>Previous</Button>}
-            {!!summary?.total_samples && <span style={{ padding: "0 10px" }}>Page {pageNumber} / {totalPageNumbers} ({summary?.total_samples} samples)</span>}
-            {summary?._links?.token && <Button disabled={isLoading} onClick={async () => {
-                searchParams.set('token', summary?._links?.token || '')
-                setSearchParams(searchParams)
-                await getProjectSummary(summary?._links?.token, true)
-            }}>Next</Button>}
-        </>
-    }
+    let table: React.ReactElement = <></>;
 
+    const totalPageNumbers = Math.ceil(
+        (summary?.total_samples || 0) / pageLimit
+    );
 
-    if (error) {
-        table = <p><em>An error occurred when fetching samples: {error}</em></p>
-    }
-    else if (!summary) {
-        if (globalContext.project) { table = <p>Loading...</p> }
-        else { table = <p><em>Please select a project</em></p> }
-    }
+    const renderLink = (s: any, k: any, external_id: string) => (
+        <Link to={`/project/${projectName}/sample/${external_id}`}>
```

Comment:
This should be the internal sample ID (`CPGxxx`)

Reply:
Fixed